processors/src/RoryClusterAnaProcessor.cxx

```diff
@@ -0,0 +1,677 @@
+        }
+        bool general = ((layer_==-1)||(module_==-1));
+        if(((LAYER==layer_)&&(MODULE==module_))||(general)){
+            //NOW IS THE PART WHERE I FILL THE CLUSTER DISTANCE HISTOGRAM
+            float Dist=69420;
+            for(int p = 0; p < Clusters_->size(); p++){ 
+                if(p==i){continue;}
+                TrackerHit * clu2 = Clusters_->at(p);
+                RawSvtHit * seed2 = (RawSvtHit*)(clu2->getRawHits().At(0));
+                float LAYER2=clu->getLayer();
+                float MODULE2=seed->getModule();
+                if((not(LAYER2==LAYER))or(not(MODULE2==MODULE))){continue;}
+                float dist = ((float)(seed2->getStrip()))-seedStrip;
+                if(dist<0){dist*=-1.0;}
+                if(dist<Dist){Dist=dist;}
+            }
+            if(Dist<69420){
+                ClusDistances_->Fill(Dist);
+                if(NTD){ClusDistancesNTD_->Fill(Dist);}
+            }
+            //std::cout<<"HELLO"<<std::endl;
+            layers_->Fill(nLayers);
+            charges_->Fill(ncharges);
+            positions_->Fill(clu->getLayer());
+            times_->Fill(ntimes);
+            if(onTrk){
+                layersOnTrk_->Fill(nLayers);
+                chargesOnTrk_->Fill(ncharges);
+                timesOnTrk_->Fill(ntimes);
+            }else{
+                layersOffTrk_->Fill(nLayers);
+                chargesOffTrk_->Fill(ncharges);
+                timesOffTrk_->Fill(ntimes);
+            }
+            if(NTD){
+                layersNTD_->Fill(nLayers);
+                chargesNTD_->Fill(ncharges);
+                positionsOnTrk_->Fill(clu->getLayer());
+                timesNTD_->Fill(ntimes);
+                if(onTrk){
+                    layersOnTrkNTD_->Fill(nLayers);
+                    chargesOnTrkNTD_->Fill(ncharges);
+                    timesOnTrkNTD_->Fill(ntimes);
+                }else{
+                    layersOffTrkNTD_->Fill(nLayers);
+                    chargesOffTrkNTD_->Fill(ncharges);
+                    timesOffTrkNTD_->Fill(ntimes);
+                } 
+            }
+        }      
+    }
+    return true;
+}
+
+void RoryClusterAnaProcessor::fillDeads(){
+    for(int i = 0;i<24576;i++){
+        Deads_[i]=0.0;
+    }
+    std::string FILENAME="/sdf/group/hps/users/rodwyer1/run/cluster_study/badchannels2021.dat";
+    std::ifstream file(FILENAME.c_str());
+    std::string line;
+    std::getline(file,line);
+    while (std::getline(file, line)) {
+        int value = std::atoi(line.c_str());
+        Deads_[value]=1.0;
+    }
+    file.close();
+    return;
+}
+
+int RoryClusterAnaProcessor::GetStrip(int feb,int hyb,int strip){
+	int BigCount = 0;
+    if(feb<=1){
+        BigCount+=feb*2048+hyb*512+strip;
+    }else{
+        BigCount+=4096;
+        BigCount+=(feb-2)*2560+hyb*640+strip;
+    }
+    return BigCount;
+}
+
+void RoryClusterAnaProcessor::Plot1(){
+    std::cout<<"I AM IN THE FINAL STEP"<<std::endl; 
```

Comment:
also, print statements like this are nice for development but should not be in the final version. If you want to keep this as a debug statement or just for keeping track of the progress of your program, please adjust it to sth along the lines of "RoryClusterAnaProcessor::Plot1:: statement here"

Reply:
Done